src/SamV71Core/SamV71Core.c

```diff
@@ -247,9 +254,40 @@ rtems_name SamV71Core_GenerateNewSemaphoreName(void)
 	return name++;
 }
 
+rtems_name SamV71Core_GenerateNewTaskName(void)
+{
+	static rtems_name name = rtems_build_name('D', 0, 0, 0);
+	return name++;
+}
+
 bool SamV71Core_SetPckConfig(const Pmc_PckId id,
 			     const Pmc_PckConfig *const config,
 			     const uint32_t timeout, ErrorCode *const errCode)
 {
 	return Pmc_setPckConfig(&pmc, id, config, timeout, errCode);
 }
+
+void SamV71Core_DisableDataCacheInRegion(void *address, size_t sizeExponent)
+{
+	assert(((uint32_t)address & (~MPU_RBAR_ADDR_MASK)) ==
+	       0); // verify proper alignment of address
+	assert(sizeExponent >= 4); // exponents less than 4 are reserved
+	assert(sizeExponent <= 31);
+	; // maximum exponent is 31 which defines 4GB region size
+	static uint8_t region = 15; // start from with highest priority
+	Mpu_RegionConfig mpuRegionConf = {
+		.address = (uint32_t)address,
+		.isEnabled = true,
+		.size = sizeExponent,
+		.subregionDisableMask = 0x00,
+		.isShareable = true,
+		.isExecutable = true,
+		.memoryType = Mpu_RegionMemoryType_StronglyOrdered,
+		.innerCachePolicy = Mpu_RegionCachePolicy_NonCacheable,
+		.outerCachePolicy = Mpu_RegionCachePolicy_NonCacheable,
+		.privilegedAccess = Mpu_RegionAccess_ReadWrite,
+		.unprivilegedAccess = Mpu_RegionAccess_ReadWrite,
+	};
+	Mpu_setRegionConfig(&mpu, region, &mpuRegionConf);
+	--region; // decrease region number, so next call will define a new one
```

Comment:
Still no assert on reaching 0. BTW, given the defaull map (could we make a reference in the comment?), I would recommend to stop at 12? I am not sure whether this limits anything, but that should give us 4 regions anyway, sufficient for 2 CANS and SpW.